crates/deno_task_shell/src/parser.rs

```diff
@@ -1449,26 +1435,100 @@ fn parse_unary_arithmetic_expr(pair: Pair<Rule>) -> Result<ArithmeticPart> {
   }
 }
 
+fn parse_unary_arithmetic_op_type(pair: Pair<Rule>) -> Result<bool> {
+  match pair.as_rule() {
+    Rule::unary_arithmetic_op => Ok(true),
+    Rule::post_arithmetic_op => Ok(false),
+    _ => Err(miette!(
+      "Unexpected rule in unary arithmetic operator: {:?}",
+      pair.as_rule()
```

Comment:
This feels fairly hacky to return true of false for the type. Maybe you can try replacing this with match statement in `unary_pre_arithmetic_expr`?